opal/mca/threads/qthreads/threads_qthreads_mutex.h

```diff
@@ -42,46 +43,67 @@
 
 BEGIN_C_DECLS
 
-typedef opal_atomic_lock_t opal_thread_internal_mutex_t;
+typedef qthread_spinlock_t opal_thread_internal_mutex_t;
 
-#define OPAL_THREAD_INTERNAL_MUTEX_INITIALIZER           OPAL_ATOMIC_LOCK_INIT
-#define OPAL_THREAD_INTERNAL_RECURSIVE_MUTEX_INITIALIZER OPAL_ATOMIC_LOCK_INIT
+#define OPAL_THREAD_INTERNAL_MUTEX_INITIALIZER           QTHREAD_MUTEX_INITIALIZER
+#define OPAL_THREAD_INTERNAL_RECURSIVE_MUTEX_INITIALIZER QTHREAD_RECURSIVE_MUTEX_INITIALIZER
 
 static inline int opal_thread_internal_mutex_init(opal_thread_internal_mutex_t *p_mutex,
                                                   bool recursive)
 {
-    opal_atomic_lock_init(p_mutex, 0);
+    opal_threads_ensure_init_qthreads();
+    #if OPAL_ENABLE_DEBUG
+    int ret = qthread_spinlock_init(p_mutex,recursive);
+    if (QTHREAD_SUCCESS != ret) {
+        opal_output(0, "opal_thread_internal_mutex_init()");
```

Comment:
This error message is not very informative. In fact none of the error messages in this PR are trying to be user friendfly.

Reply:
Also: can we use proper `opal_show_help()`-style help messages?  That allows you to do two things:

1. Easily write good help messages (because you write them in a text file, not a series of printf-like C calls)
2. Error messages will automatically be de-duplicated (so that the user is not spammed with the same error message from each MPI process)

Ping me off-list if you want a pointer in the right direction for how to use `opal_show_help()`.

Reply:
`opal_show_help()` looks very useful. We use the old output throughout that MCA module. I would prefer to update the debug output and verbosity in another PR.

Reply:
How about a compromise: don't use `opal_output()` in this PR -- use `opal_show_help()` for all the places where you added new `opal_output()` statements in this PR.  Don't worry about fixing the entire component; you can do that in a future PR.

I say this simply because allowing this PR would be increasing technical debt.  Might as well make the content of this PR be good / not incur technical debt.  I agree that a future PR can fix up the *rest* of the component.

Reply:
ok